network/validator/authorized_sender_validator.go

```diff
@@ -105,16 +105,24 @@ func (av *AuthorizedSenderValidator) Validate(from peer.ID, msg interface{}, cha
 //
 // Expected error returns during normal operations:
 //   - ErrSenderEjected: if identity of sender is ejected from the network
-//   - message.ErrUnknownMsgType if message auth config us not found for the msg
+//   - codec.ErrUnknownMsgCode: if interface for the encoded message code byte is unknown
+//   - message.UnknownMsgTypeErr if message auth config us not found for the msg
 //   - message.ErrUnauthorizedMessageOnChannel if msg is not authorized to be sent on channel
 //   - message.ErrUnauthorizedRole if sender role is not authorized to send msg
-func (av *AuthorizedSenderValidator) isAuthorizedSender(identity *flow.Identity, channel channels.Channel, msg interface{}, protocol message.Protocol) (string, error) {
+func (av *AuthorizedSenderValidator) isAuthorizedSender(identity *flow.Identity, channel channels.Channel, msgCode uint8, protocol message.Protocol) (string, error) {
```

Comment:
Instead of using `uint8` variables to represent message codes throughout the codebase, let's define the `MessageCode` type. This will make it clear to anyone reading the code that a variable of this type represents a message code, and will also prevent any potential bugs caused by accidentally using a non-message code value. In this way, we take advantage of the type system in Go to improve the readability and maintainability of the code.

```go
type MessageCode uint8
```

Reply:
https://github.com/onflow/flow-go/pull/3831/commits/938612350daa63faf74941be04e42211fc6c9a7e